Replace the hostname in the fluentd config file even if the file exists

barney-s opened the pull request:
Startup script did not replace the hostname when the image come preinstalled with stackdriver agent (config file exists). This caused the wrong hostname (hostname at buildtime) to show up in the stackdriver logs.

**What type of PR is this?**
/kind bug

**What this PR does / why we need it:**
This PR fixes stackdriver config to have the correct hostname.

**Which issue(s) this PR fixes:**

**Special notes for your reviewer:**

**Does this PR introduce a user-facing change?:**
No

barney-s committed:
Replace the hostname in the fluentd config file even if the file exists

barney-s commented:
/assign @pjh

RA489 commented:
/release-note-none

RA489 commented:
/ok-to-test

pjh commented on cluster/gce/windows/k8s-node-setup.psm1:
```diff
@@ -1528,13 +1528,12 @@ function Configure-LoggingAgent {
   if (-not (ShouldWrite-File $fluentd_config_file)) {
```
Instead of adding the else block, can you just remove this `if (ShouldWrite-File ...)` block entirely? Then I'm guessing that all the code below can stay the same and the file will be written / overwritten every time.

barney-s committed:
Removing conditional check 

Addressing review comment. Removing conditional check for fluentd config file path.

pjh commented:
/lgtm
/approve

yliaog commented:
/retest

yliaog commented:
/retest
